dali/util/nvml.h

```diff
@@ -28,6 +28,18 @@
 #include "dali/core/cuda_utils.h"
 #include "dali/util/nvml_wrap.h"
 
+
+#define NVML_CALL(code)                                    \
```

Comment:
Perhaps it would be better to have a specialized exception rather than throw it into the `DALI_FAIL` bin.
You can specialize cudaResultCheck, like it's done for cuFFT, and use CUDA_CALL to check the result.

Reply:
Done